examples/instance_refresh/main.tf

```diff
@@ -0,0 +1,259 @@
+      ]
+      "resources" : [
+        module.eks.workers_asg_arns[0],
+      ]
+    }
+  )
+}
+
+resource "aws_cloudwatch_event_target" "node_term_event_target" {
+  rule      = aws_cloudwatch_event_rule.node_term_event_rule.name
+  target_id = "ANTHandler"
+  arn       = module.node_term_sqs.sqs_queue_arn
+}
+
+module "node_term_role" {
+  source                        = "terraform-aws-modules/iam/aws//modules/iam-assumable-role-with-oidc"
+  version                       = "4.1.0"
+  create_role                   = true
+  role_description              = "IRSA role for ANTH, cluster ${local.cluster_name}"
+  role_name_prefix              = local.cluster_name
+  provider_url                  = replace(module.eks.cluster_oidc_issuer_url, "https://", "")
+  role_policy_arns              = [aws_iam_policy.node_term.arn]
+  oidc_fully_qualified_subjects = ["system:serviceaccount:${var.namespace}:${var.serviceaccount}"]
+}
+
+resource "null_resource" "helm_install" {
+  provisioner "local-exec" {
+    environment = {
+      USE_SUDO         = "false"
+      HELM_INSTALL_DIR = "${path.root}/bin"
+    }
+    command = <<EOT
+mkdir -p ${path.root}/bin
+curl -fsSL -o ${path.root}/bin/get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
+chmod +x ${path.root}/bin/get_helm.sh
+${path.root}/bin/get_helm.sh
+EOT
+  }
+}
+
+resource "null_resource" "helm_add_repo" {
+  depends_on = [
+    null_resource.helm_install,
+  ]
+
+  triggers = {
+    version = var.aws_node_termination_handler_chart_version
+  }
+
+  provisioner "local-exec" {
+    environment = {
+      XDG_CACHE_HOME  = abspath("${path.root}/.cache")
+      XDG_CONFIG_HOME = abspath(path.root)
+      XDG_DATA_HOME   = abspath(path.root)
+    }
+    command = "${path.root}/bin/helm repo add --force-update eks https://aws.github.io/eks-charts"
+  }
+}
+
+resource "null_resource" "helm_install_chart" {
+  depends_on = [
+    module.eks,
+    module.node_term_role,
+    null_resource.helm_add_repo,
+  ]
+
+  triggers = {
+    version  = var.aws_node_termination_handler_chart_version
+    role_arn = module.node_term_role.iam_role_arn
+  }
+
+  provisioner "local-exec" {
+    environment = {
+      KUBECONFIG      = "${path.root}/${module.eks.kubeconfig_filename}"
+      XDG_CACHE_HOME  = abspath("${path.root}/.cache")
+      XDG_CONFIG_HOME = abspath(path.root)
+      XDG_DATA_HOME   = abspath(path.root)
+    }
+    command = <<EOT
+chmod 0600 "$KUBECONFIG"
+${path.root}/bin/helm upgrade --install aws-node-termination-handler \
+  --wait \
+  --namespace ${var.namespace} \
+  --set serviceAccount.name=${var.serviceaccount} \
+  --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${self.triggers.role_arn} \
+  --set enableSqsTerminationDraining=true \
+  --set queueURL=${module.node_term_sqs.sqs_queue_id} \
+  --set logLevel=DEBUG \
+  --set awsRegion=${var.region} \
+  --version=${self.triggers.version} \
+  eks/aws-node-termination-handler
+EOT
+  }
+}
```

Comment:
Why can't we use the [helm terraform provider](https://registry.terraform.io/providers/hashicorp/helm/latest/docs) to install the node termination handler ?

Reply:
We tried it in the past and had some issues with it. Perhaps those are issues no longer exist. I will give it a try.

Reply:
please let me know, I would like to merge this tonight or tomorrow.

Reply:
Please to it tonight. Thanks :-)

Reply:
I can confirm that the latest Helm provider can install the _aws-node-termination-handler_, we've had no issues with this at all.

Reply:
Yes, it's working fine now. I wanted to confirm that destroy ran cleanly.

Reply:
Pushing the requested changes soon.